### ISSUE TITLE
Add convenience wrapper for cancelling outstanding activities or document best practice

### ISSUE THREAD

**Issue by ryanwalls** [sclasen/swfsm#117]
Currently trying to implement a scenario where `OnWorkflowCancelRequested`, all outstanding activities are requested to cancel, and then wait for those cancellations to process before then cancelling the workflow.  

I've implemented cancelling all outstanding activities like this:  

```
fsm.OnWorkflowCancelRequested(fsm.AddDecisions(cancelOutstandingActivities), fsm.Transition("waitForActivityCancellations"))
```

Where `cancelOutstandingActivities` looks like this:
```
var cancelOutstandingActivities = typed.MultiDecisionFunc(func(f *fsm.FSMContext, h *swf.HistoryEvent, d *StateData) []*swf.Decision {
	decisions := f.EmptyDecisions()
	for _, pendingActivity := range f.ActivitiesInfo() {
		cancelActivityDecision := &swf.Decision{
			DecisionType: aws.String(swf.DecisionTypeRequestCancelActivityTask),
			RequestCancelActivityTaskDecisionAttributes: &swf.RequestCancelActivityTaskDecisionAttributes{
				ActivityId: aws.String(pendingActivity.ActivityId),
			},
		}
		decisions = append(decisions, cancelActivityDecision)
	}
	return decisions
})
```

I then transition to `waitForActivityCancellations` where I'm waiting for all activities to either timeout, fail, complete, or cancel before I cancel the workflow.  But knowing when they are all done is challenging.  When events come in that would remove the activity from the correlator, they are not processed until after the event is seen by my decider.  Thinking about handling this with an interceptor that fires after the decider...  

Currently have hacked around it by listening to activity timeouts, fails, completes and cancels, and if there is only 1 activity left in the correlator, verifying that it matches the incoming event.  

Is there a better way to handle cancelling all outstanding activities?

**Comment by sclasen** [comment]
This approach is reasonable, and we have done similar in at least one case here. 

The other way to do it is to add some flags to your data struct, and flip them as you see the cancelations/fails/completes coming


```go
type MyData struct {
   OtherStuff interface{}
   ActivityOneDone bool
   ActivityTwoDone bool
}

func allDone(data *MyData) bool {
   return data.ActivityOneDone && data.ActivityTwoDone
}

OnData(typed.PredicateFunc(allDone), fsm.CompleteWorkflow())

```

**Comment by ryanwalls** [comment]
@sclasen Thanks.  What do you think about using a `DecisionInterceptor.AfterDecision` instead?  Wouldn't the activities list in the correlator be cleaned up by that point?

**Comment by ryanwalls** [comment]
Also any thoughts on making a decider that handles this case as part of the library?  If you think it's a good approach, I could probably take a crack at it.

**Comment by sclasen** [comment]
@ryanwalls yep and AfterDecision interceptor would work, the correlator would be cleaned up.

Sure, have a crack at that decider, no harm in including such a thing.
